### ISSUE TITLE
Client/VM is stuck at 1024x768

### ISSUE THREAD

**Issue by kingkj52** [jamesstringerparsec/Easy-GPU-PV#190]
I've followed the process (have a 3080 Ti), but when connecting in, the VM, in windows display settings, is locked to 1024x768. Even after using the update drivers ps1 file, and using the file to gather driver files into a zip, manually copying them over to C:\Windows, and rebooting, the issue remains. It is acting as if the client has no graphics drivers. Is there a workaround for this, or did I do something wrong?

**Comment by Swany6mm** [comment]
I don't have a solution, but I've noticed that when you disable the MS Hyper V Adapter, it defaults to an MS Basic Adapter, which can't go any higher. (validate by right clicking Desktop>Display>Advanced Setting).

As of now, I cannot get Parsec to go above 1080p no matter what I do because of this. MS Basic is limited to 768p and the Hyper Visor Display is limited to 1080. I can SEE the nVidia adapter is Device Manager, but other than when I use RDP, I have never been able to get Parsec to go above 1080p (the dummy plug I have goes well above that and works with other Streaming Services)

**Comment by iwolfking** [comment]
I am not sure the exact cause, but I was able to use [Custom Resolution Utility](https://www.monitortests.com/forum/Thread-Custom-Resolution-Utility-CRU) to add a custom resolution and then was able to set the new resolution in Display Settings.

The VM originally was not locked to 1024x768, but after a Windows Update and restart, that is when the issue started occurring.

**Comment by Swany6mm** [comment]
I tried that out, and after doing some reading in the forums, it wont help. It comes back around to the VM only seeing the Hyper V Video and not the Parsec Video or the actual nVidia GPU, and it can't make changes to the Hyper V setup. I'm starting to toy with the idea of giving VMWare ESXi a try since nVidia opened up their drivers earlier this year. See if that can't get it going since it's a bit more mature regarding GPU sharing/partitioning and Passthrough

**Comment by klemens-st** [comment]
This issue is easily fixed by installing a virtual display like [usbmmidd](https://www.amyuni.com/forum/viewtopic.php?t=3030)

The cause of the issue you are experiencing is that recent versions of Parsec free edition launch Parsec virtual display only if no monitors are attached while adding virtual displays at will is reserved for paid editions. We have two problems here:
1. Hyper-V always plugs in its virtual display which does not support hardware acceleration, i.e. Parsec virtual display is never activated because there *always is* a display. Current release of Hyper-V does that even if you disable Hyper-V display in the guest.
2. When no displays which support Nvidia rendering are connected, GPU driver disables acceleration, and the resolution is capped at 1024x768

The solution is to install a virtual display that is always "plugged in", like the one I mentioned above. This is analogous to using a dummy HDMI plug. You should then set the guest to display only this new virtual display.

**Comment by Swany6mm** [comment]
I found a band-aid/work around that is working. Don't like it as a perminant solution, so I'll keep researching, but will use this for know. Apologies, I don't knokw the proper way to link in git. I'll put a link to the thread and the actual message I used.

link
https://github.com/jamesstringerparsec/Easy-GPU-PV/issues/59


"Ok I think I found a workaround, the trick is to activate a virtual display before the parsec driver.
the VM is running windows 11 been access using windows 10.
All the steps on the VM:
    Connect (parsec / hyper-v).
    Download and install [usbmmidd_2](https://www.amyuni.com/forum/viewtopic.php?t=3030) : this is a virtual display that we can control, take a look at the instructions, pretty simple.
    Activate using "deviceinstaller64 enableidd 1": Now you will have a virtual monitor.
    Create a task on task manager to run a bat file/command activating the virtual monitor: "deviceinstaller64 enableidd 1" and set it to run on windows start up and admin privileges without the need of the user to be logged in (after creating the task check for advanced settings or something like that).
    Disable the driver display of hyper-v (if its not already) on device manager.
    Reboot VM
    On display settings, disable all monitors that are not using the parsec driver (tip: take a look at refresh rate, parsec is the only one that goes above 60).
    Just to be sure, reboot VM again to check if the settings are being saved.
    Have fun.

PS: even the option to use client resolution is working."

**Comment by klemens-st** [comment]
Yes, usbmmidd needs to be enabled on each boot. I did it by adding a startup script in `gpedit.msc` but other methods work too.

**Comment by veazel** [comment]
Well the actual issues is . 

When I log into my vm through the same account that is logged into my vm persecs, the persec virtual display Adapter started working, and it shows me the correct resolution ie. 1080p or even allow me to change the resolution of my vm. 

But as soon as I disconnect and try to log in from guest persec account.  Persec virtual display adapter  didn't work. And Microsoft basic display driver seems to be on i.e. 1024x768 resolution gets displayed. 

So the actual problem is how persec virtual display Adapter acts on host and guests connect.  

Usbmmidd display driver works only if u uninstall and re-install it again and again after each remote session, which is quite annoying. 

I hope this makes the issue clear , And help in more trouble shooting.  

Also a very thanks for providing this script, can't say it's perfect but still it do open many opportunities.

I hope this issue with persec virtual display adapter will get fixed soon.  

Thanks again.

**Comment by gdias1992** [comment]
I had to create a powershell script to make it work:

**startup.ps1**
`C:\usbmmidd_v2\deviceinstaller64 enableidd 1`

In the Task Scheduler create a task that:

- Run whether user is logged or not
- Run with the highest privileges
- Run at startup
- Deactivate the conditions
- Has the following actions:

1. Program/script: powershell.exe
2. Add arguments: -NoExit -ExecutionPolicy Bypass C:\usbmmidd_v2\startup.ps1


**Edit the paths to match your system.**

Enjoy.

**Comment by Joekraft** [comment]
@gdias1992 thanks for the clear instructions

I'll just add for anyone following them to also be sure to run _usbmmidd.bat_ after downloading USBMMIDD to install the interfaces.

**Comment by mad-ady** [comment]
I too had the same issue after deploying a new Win 10 VM. The suggestions in this thread helped solve this issue, but - although I can see Nvidia 2060 in the Device Manager (working ok), and now I can change screen resolution, I can't see the GPU section in Task Manager, dxdiag says there is no d3d renderer, and UserBenchmark also fails with no d3d renderer. Any ideas what else I'm missing?

**Comment by klemens-st** [comment]
As long as GPU acceleration works without issues in your VM you can ignore the problems you mentioned. It is because of how Windows handles partitioned GPUs currently and should not affect performance. But yeah no GeForce Experience or other stuff that normally works with physical cards will work in the VM.

**Comment by klemens-st** [comment]
If anyone is still using my solution from one of the [previous comments](https://github.com/jamesstringerparsec/Easy-GPU-PV/issues/190#issuecomment-1240367095): I was not right about virtual displays no longer working on free editions of Parsec. What happened is that when they added the fallback feature, previous virtual display setting was auto removed from the config file. But is still works if you add it back *manually* to the config file as explained in #204

So no need to use usbmmidd after all :D

**Comment by augustorubiao** [comment]
I've tried all the above, had some issues and started tinkering the Hyper-V myself, found enhanced session mode activ on user section according to the scresnhots below.

The Hyper-V driver came back to life right after I deactivated it, accepting all resolutions.

Hope this does help someone.

![image](https://user-images.githubusercontent.com/56486400/212779284-51c6ab55-225c-4c6b-a280-5a4d7f74fc8c.png)

![image](https://user-images.githubusercontent.com/56486400/212779317-604938ca-73ca-4345-be3f-96f3619f2a01.png)

**Comment by totalitarian** [comment]
I'm confused what the solution is here. I get 1024x768 in parsec. Should we be editing a config, running a .ps, installing an external driver or something else?

Any help would be appreciated thanks :)

**Comment by augustorubiao** [comment]
I've ended up using third-party custom resolution software along with a simple batch file running on Windows start to add the custom resolution + disable the Hyper-V Video driver to avoid BSOD.

Unfortunately I am not using the virtual PC right now since I have been having personal issues with RAM. I will try to grab this content for you.

**Comment by mad-ady** [comment]
> I'm confused what the solution is here. I get 1024x768 in parsec. Should we be editing a config, running a .ps, installing an external driver or something else?
> 
> Any help would be appreciated thanks :)

Sorry for not having the time to reply sooner.
I'm currently using the usbhimdd_v2 solution discussed above. The display is activated on logon via a scheduled task. I do have resolution problem when connecting via parsec -one instance comes up at 1024x768. So I applied a second "fix". I downloaded a command-line utility called qres.exe that allows me to set the resolution via script - something like `qres.exe /x:1920 /y:1080`. I do start it with a 5s delay after enabling the display, but it doesn't always work. So I keep it as a desktop shortcut handy (for various resolutions) that I activate manually. The downside is - if I disconnect and reconnect from parsec, it reverts to 1024x768, and I need to run the script manually.

**Comment by klemens-st** [comment]
Usbmmidd is not needed as I mentioned earlier. See [this comment](https://github.com/jamesstringerparsec/Easy-GPU-PV/pull/204#issuecomment-1252096353) and the discussion in that PR.

Let us know if you followed the instructions and added the virtual display config line manually but it still didn't work.

**Comment by MungbeanOuO** [comment]
> Usbmmidd is not needed as I mentioned earlier. See [this comment](https://github.com/jamesstringerparsec/Easy-GPU-PV/pull/204#issuecomment-1252096353) and the discussion in that PR.
> 
> Let us know if you followed the instructions and added the virtual display config line manually but it still didn't work.

Just tested it, and this tweak didn't work.

**Comment by klemens-st** [comment]
> > Usbmmidd is not needed as I mentioned earlier. See [this comment](https://github.com/jamesstringerparsec/Easy-GPU-PV/pull/204#issuecomment-1252096353) and the discussion in that PR.
> 
> > 
> 
> > Let us know if you followed the instructions and added the virtual display config line manually but it still didn't work.
> 
> 
> 
> Just tested it, and this tweak didn't work.

Share your config file

**Comment by MungbeanOuO** [comment]
> Share your config file

I removed it in User\Install.ps1 according to that article
$configfile += "host_virtual_monitors = 1"
$configfile += "host_privacy_mode = 1"
Then added
$configfile += "host_virtual_monitor_fallback = 1"
![image](https://user-images.githubusercontent.com/42833513/221361689-6e5ed39b-a32e-4218-b65c-0260e58e1516.png)

After that recreate a new VM.
After the VM restarts, it returns to 1024x768. I have confirmed that the parameters in C:\ProgramData\Parsec\config.txt still exist. 
![image](https://user-images.githubusercontent.com/42833513/221361745-f44f349f-d096-4453-80ae-06a4696362b3.png)

I have also tried closing parsec and restarting them after closing parsec.exe from the task manager, but the situation remains the same. 
Currently using usbmmidd_2 to connect through parsec, but I don't know how to adjust the refresh rate, which is a pity.

**Comment by klemens-st** [comment]
Ok try the following:
- Disable usbmmidd
- Stop parsec and all its processes 
- Remove `host_virtual_monitor_fallback = 1`
- start parsec and try to connect again

**Comment by klemens-st** [comment]
This has to be done ***after*** the vm is created. The config has to include `host_virtual_monitors = 1`

**Comment by Kodikuu** [comment]
Do not modify the configs set by the script, they are required.

**Comment by klemens-st** [comment]
@Kodikuu Im referring to [your answer to my question](https://github.com/jamesstringerparsec/Easy-GPU-PV/pull/204#issuecomment-1252096353) which has worked for me fine. It should also work for others in this thread.

**Comment by Kodikuu** [comment]
Yes, in the config itself, after setup of a VM that no longer has that setting set.

The script already sets this, and host_privacy_mode. Both of which are needed. Do not modify the configs set by the script.

The thread you linked was a test of a new setting which failed.

**Comment by MungbeanOuO** [comment]
The script was changed back and it finally worked, but there are many details that need to be done to make the virtual display driver work properly. Or maybe my computer is weird.
1.
After the Windows update in the VM, I encountered the problem that I could not log in automatically and the password was wrong. I don’t have a better solution for this, so I just re-created a VM.
2.
When the VM enters the desktop for the first time, the browser or a certain application screen (for example, discord) on the physical machine may freeze. If you interact with that window, the entire screen will freeze. My solution is to re-plug the display output of the physical machine .
3.
The host_virtual_monitors = 1 and host_privacy_mode = 1 entered by the script may disappear from the config when connecting to parsec for the first time, and they need to be added together with host_virtual_monitor_fallback = 1 to call out the virtual monitor normally.
I don't know if I should update the VM's system, but I'm used to doing it. However, the update and restart will conflict with the parsec virtual display driver and config, which will cause the shutdown to be stuck or the first point.
This part still needs more testing to see what the correct order of execution is.
4.
The parsec remote screen cannot operate the operating system application (such as task manager).
My solution is to close the parsec application and restart the VM.

The above are the problems and solutions I generally encountered during the setting process. My English is not good, so I used machine translation for my reply. Please forgive me if there are any typos.
![image](https://user-images.githubusercontent.com/42833513/221395937-a1317409-c6d1-405e-9816-444321b291c2.png)

**Comment by H4M5TER** [comment]
This happened when I connect the VM using another Parsec account, which means the privacy mode can't be enabled and virtual monitor won't be created.
I tried the usbmmidd solution but it's not good enough.

**Comment by klemens-st** [comment]
@MungbeanOuO

> After the Windows update in the VM, I encountered the problem that I could not log in automatically and the password was wrong. I don’t have a better solution for this, so I just re-created a VM.

It happened to me only once, no idea what the reason was. Never happened again.

> The host_virtual_monitors = 1 and host_privacy_mode = 1 entered by the script may disappear from the config when connecting to parsec for the first time, and they need to be added together with host_virtual_monitor_fallback = 1 to call out the virtual monitor normally.

Yes, this is exactly what I was referring to in my previous comment, this is what @Kodikuu pointed me to in [another thread](https://github.com/jamesstringerparsec/Easy-GPU-PV/pull/204#issuecomment-1252096353) but it happens to be the correct solution for the problem in this thread.

> The parsec remote screen cannot operate the operating system application (such as task manager).

@mikekachar In this thread we are helping other make this work in Hyper-V (which it does, without problems if set up correctly)
My solution is to close the parsec application and restart the VM.

Strange, the VM installs parsec on the system, not per user. I never encountered this problem.

**Comment by db00t** [comment]
I activated Microsoft Hyper-V Video in device manager on VM and got my 1920x1080 resolution back.

**Comment by lucasfirl** [comment]
> 

This fixed it for me!

**Comment by CarbonicKevin** [comment]
For a fix that worked for me:

https://github.com/jamesstringerparsec/Easy-GPU-PV/issues/59#issuecomment-1615955872

**Comment by ColdCactus** [comment]
> I activated Microsoft Hyper-V Video in device manager on VM and got my 1920x1080 resolution back.

Bam, fixed! Awesome!

I don't know for sure why it got deactivated, but I might have forgotten to keep Guest Services turned off in hopes of fixing some strange cursor lag (only on the physical computer, not via parsec...).

**Comment by timminator** [comment]
I came across this problem aswell. Using the hyper-v adapter as mentioned in other posts was not an option for me due to its resolution and refresh rates limits.
usbmmid_v2 was also not a viable option due to its lack of hardware cursor support causing a double cursor when connecting through Parsec.
So i decided to improve this project and created my own fork. You can find it under the following link:

https://github.com/timminator/Enhanced-GPU-PV

It now adds a virtual display on startup to the VM by taking advantage of another  [project](https://github.com/timminator/ParsecVDA-Always-Connected)) of mine and it will stay connected until your next shutdown/restart.
The virtual display is also based on the Parsec Virtual Display Driver, which means that the advantages of a high resolution and high refresh rate up to 4K@240Hz can still be utilized. 
This resolves the aforementioned issues and problems raised in this thread without having to make any compromises.

If you're interested, try it out and if you're happy with it, i would appreciate a star rating :-)

I hope other people will find it valuable aswell!

**Comment by timminator** [comment]
> Hi, is your project supporting HDR on windows 11? Thank you

Unfortunately, the Parsec Virtual Display Driver used in this project does not currently support HDR.

**Comment by timminator** [comment]
I updated my [project](https://github.com/timminator/Enhanced-GPU-PV) mentioned in a post before. It now allows the user to also install the Virtual Display Driver with HDR. It also adds the option to use Sunshine instead of Parsec.
If anyone is interested, please check it out!

**Comment by jamesstringer90** [comment]
Hey, just so you know, Easy-GPU-PV is being archived, and replaced with [App Sandbox](https://github.com/jamesstringer90/appsandbox)

It should be a lot easier to use, as it's a full app and not a collection of Powershell scripts.

And, most importantly, still completely free and open source!

Some features:

- Full App with interface for creating App Sandbox VMs
- App Sandbox runs on Windows 11 Home and Pro
- Supports creating Windows 11 or Ubuntu 26.04 LTS VMs
- Supports GPU-PV with much better support for OpenGL and better support for Vulkan
- Much better at different Language virtual machines
- Copy and Paste Support
- No need for third party game streaming apps (you can still use them though)
- Can create VMs without internet connections and they'll still work great
- Portable App, run from your Zip 
 
[Hopefully you get to try it out!](https://github.com/jamesstringer90/appsandbox/releases/tag/v0.1.0)
